src/braket/circuits/circuit.py

```diff
@@ -643,6 +657,69 @@ def add_verbatim_box(
             self._has_compiler_directives = True
         return self
 
+    def measure(self, target_qubits: Optional[np.ndarray] | Optional[int] = None) -> Circuit:
+        """
+        Add a `measure` operator to `self` ensuring only the target qubits are measured.
+
+        Args:
+            target_qubits (Optional[ndarray] | Optional[int]): target qubits to measure.
+                Default=None
+
+        Returns:
+            Circuit: self
+
+        Raises:
+            IndexError: If `circuit` has no qubits.
+            IndexError: If target qubits are not within the range of the current circuit.
+
+        Examples:
+            >>> circ = Circuit.h(0).cnot(0, 1).measure([0])
+            >>> circ.print(list(circ.instructions))
+            [Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(0)]),
+            Instruction('operator': CNot('qubit_count': 2), 'target': QubitSet([Qubit(0),
+                Qubit(1)]),
+            Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(2)]),
+            Instruction('operator': Measure, 'target': QubitSet([Qubit(0)])]
+        """
+        # check whether measuring an empty circuit
+        if not self.qubits:
+            raise IndexError("Cannot measure an empty circuit.")
+
+        if type(target_qubits) is int:
```

Comment:
To pick up subclasses of `int`: 
```suggestion
        if isinstance(target_qubits, int):
```
(but this is likely irrelevant if using `QubitSetInput` for `target_qubits`)